### PR TITLE
feature: typescript-axios support single request param

### PR DESCRIPTION
**Pull request by erdembas**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [X] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [X] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [X] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [X] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java [hunk 1]
```diff
@@ -164,8 +172,8 @@ public void processOpts() {
 
     @Override
     protected void addAdditionPropertiesToCodeGenModel(CodegenModel codegenModel, Schema schema) {
-        codegenModel.additionalPropertiesType = getTypeDeclaration(ModelUtils.getAdditionalProperties(schema));
-        addImport(codegenModel, codegenModel.additionalPropertiesType);
+    codegenModel.additionalPropertiesType = getTypeDeclaration(ModelUtils.getAdditionalProperties(schema));
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java:175]: @erdembas thanks for the PR. Prefer to keep the original code indentation.

File: modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache [hunk 2]
```diff
@@ -131,17 +161,43 @@ export const {{classname}}AxiosParamCreator = function (configuration?: Configur
...
+                    let mapped = {{paramName}}.map(value => (<any>"{{{dataType}}}" !== "Array<string>") ? JSON.stringify(value) : (value || ""));
+                    localVarHeaderParameter['{{baseName}}'] = mapped.join(COLLECTION_FORMATS["{{collectionFormat}}"]);
+                    }
+                {{/useSingleRequestParameter}}
+                {{#useSingleRequestParameter}}
+                    if (requestParameters.{{paramName}}) {
+                    let mapped = requestParameters.{{paramName}}.map(value => (<any>"{{{dataType}}}" !== "Array<string>") ? JSON.stringify(value) : (value || ""));
+                    localVarHeaderParameter['{{baseName}}'] = mapped.join(COLLECTION_FORMATS["{{collectionFormat}}"]);
+                    }
+                {{/useSingleRequestParameter}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache:197]: @erdembas I wonder if you can revise the code indentation in the above code block. Thanks.

Comment (erdembas) [reply modules/openapi-generator/src/main/resources/typescript-axios/apiInner.mustache]: @wing328 this block indent edited.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java (hunk 1 above)
Comment (erdembas) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java]: @wing328 edited this block. Thanks.

**Comment by erdembas** [comment]
Why does the circle-ci pipeline fail? Everything looks smoothly.. @wing328

**Comment by macjohnny** [comment]
@erdembas can you please fix the conflicts?

**Comment by macjohnny** [comment]
this feature was already implemented with https://github.com/OpenAPITools/openapi-generator/pull/6288

**Pull request closed without merging**